tests/test_storage.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+import boost_histogram as bh
+import numpy as np
+from numpy.testing import assert_array_equal
+
+
+@pytest.mark.parametrize(
+    "storage",
+    [bh.storage.Int, bh.storage.Double, bh.storage.AtomicInt, bh.storage.Unlimited],
+)
+def test_setting(storage):
+    h = bh.Histogram(bh.axis.Regular(10, 0, 1), storage=storage())
+    h[bh.underflow] = 1
```

Comment:
this looks like you are testing both uhi and the storage. i think it would be better to not also use uhi here.